src/ComposerRequireChecker/FileLocator/LocateFilesByGlobPattern.php

```diff
@@ -11,10 +11,10 @@ class LocateFilesByGlobPattern
     /**
      * @param string[] $globPatterns a list of glob patterns to find files in
      * @param string $rootDir the root directory that should be used when patterns are relative paths
-     * @return iterable|string[] the files found by the given glob patterns
+     * @return Generator the files found by the given glob patterns
```

Comment:
I suggest keeping these as `iterable`: don't want people to call `Generator` methods on this result (especially not `Generator#send()` :D )

Let's also document that this is an `iterable<???, string>`